### ISSUE TITLE
Issue Report - Xttsv2.convert(): Got unsupported ScalarType BFloat16

### ISSUE THREAD

**Issue by senderic** [DrewThomasson/ebook2audiobook#1907]
1)**Operating System:**
 - Windows, Linux (WSL2)

2)**Hardware and Model**
 - CUDA
 - Nvidia RTX Pro 3500

3)**Script Mode**
- Podman Compose

```yml
version: "3.8"

services:
    ebook2audiobook:
        image: athomasson2/ebook2audiobook:cu128
        container_name: ebook2audiobook
        restart: always
        ports:
            - "7860:7860"
        entrypoint: ["bash", "ebook2audiobook.command", "--script_mode", "full_docker"]
        tty: true
        stdin_open: true
        devices:
            - nvidia.com/gpu=all
        volumes:
            - ./ebooks:/app/ebooks:rw
            - ./audiobooks:/app/audiobooks:rw
            # - ./models:/app/models:rw
            # - ./voices:/app/voices:rw
            - ./tmp:/app/tmp:rw
        environment:
            - CUDA_VISIBLE_DEVICES=0
            - FORCE_CUDA=1
            - NVIDIA_VISIBLE_DEVICES=all
            - NVIDIA_DRIVER_CAPABILITIES=compute,utility
            - TOKENIZERS_PARALLELISM=false
            - NCCL_P2P_DISABLE=0
            - NCCL_DEBUG=WARN
            - WORKSPACE=/workspace
            - PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True
            - TZ=America/Los_Angeles
            - LOGLEVEL=DEBUG
        shm_size: "8g"     # shared memory for PyTorch DDP tensor sharing
        ulimits:
            memlock: -1       # allow memory locking (required by some CUDA operations)
            stack: 67108864   # 64 MB stack
```
 
4)**Full Terminal log As Attached File**
- from the command line run to start E2A until the end. Attach the log file, don't copy/paste it as a comment.

[run1.log](https://github.com/user-attachments/files/28661185/run1.log)

```
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.210                Driver Version: 573.96         CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA RTX 3500 Ada Gene...    On  |   00000000:01:00.0 Off |                  Off |
| N/A   50C    P8              5W /  110W |    1989MiB /  12282MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A              38      C   /python3.12                           N/A      |
+-----------------------------------------------------------------------------------------+
```

**We don't need other comments or suggestions. We will ask to you if we need it thanks.

**Comment by DrewThomasson** [comment]
Try the file replace solution said here

https://github.com/DrewThomasson/ebook2audiobook/issues/1906#issuecomment-4632380247

**Comment by senderic** [comment]
That worked! Thank you!

**Comment by webmaxru** [comment]
For anyone landing here from a fresh install: this isn't actually fixed in the shipped code yet. The only fix so far is the manual `lib/classes/tts_engines/common/utils.py` swap Drew posted in #1906, which works per-user but isn't in the release. The latest release (v26.6.9) still crashes out-of-the-box on CUDA GPUs with compute capability ≥ 8 (Ampere / Ada / Blackwell).

Root cause:

- `_apply_gpu_policy()` in `lib/classes/tts_engines/common/utils.py` still selects a `torch.bfloat16` autocast for `cc_major >= 8`. (It was reported as removed in v26.5.20, but it's still there in v26.6.9.)
- `XTTS.convert()` runs `engine.inference(...)` inside that autocast.
- Coqui's `Xtts.inference()` finishes with `torch.cat(wavs, dim=0).numpy()` (`TTS/tts/models/xtts.py`, line 561). NumPy has no `bfloat16` dtype, so it raises `Got unsupported ScalarType BFloat16`.

The MPS path avoids this because it uses `float16` (which NumPy supports), not `bfloat16`. Same trace was also reported still occurring on an RTX 5060 Ti in #1793.

I have a small fix that keeps bf16 for the GPT compute and only upcasts to float32 at the `.numpy()` boundary — happy to open a PR if that's useful.

**Comment by ROBERT-MCDOWELL** [comment]
this issue depends on CUDA gpu hardware... some supports it, some not. it's almost impossible to select bf16 on case by case, so next release I will replace it with f16.

**Comment by Neltherion** [comment]
is this fixed in the new release? can the issue be closed?

**Comment by ROBERT-MCDOWELL** [comment]
it is closed already and yes it should be ok now.

**Comment by Neltherion** [comment]
I tried pulling a newer version of this Docker and running it but the problem still remains. which docker image version should I try? right now I'm running into the problem with this:

docker run -v "./ebooks:/app/ebooks" -v "./audiobooks:/app/audiobooks" -v "./models:/app/models" -v "./voices:/app/voices" -v "./tmp:/app/tmp" --gpus all --rm -it -p 7860:7860 athomasson2/ebook2audiobook:cu128

**Comment by ROBERT-MCDOWELL** [comment]
check the version of e2a. it should be v26.7.23
it's not possible you still have bf16 scalar error again since I completely removed it from the code.
you should rebuild the docker image and delete all the docker cache.
